multicluster/controllers/multicluster/commonarea/remote_common_area_test.go

```diff
@@ -22,21 +22,84 @@ import (
 	"time"
 
 	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog/v2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 
 	mcsv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
-	"antrea.io/antrea/multicluster/test/mocks"
-)
 
-var (
-	fakeRemoteClient = fake.NewClientBuilder().WithScheme(scheme).Build()
+	"antrea.io/antrea/multicluster/test/mocks"
 )
 
 func TestMemberAnnounce(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	mockManager := mocks.NewMockManager(mockCtrl)
+	fakeRemoteClient := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+	remoteCommonAreaUnderTest := &remoteCommonArea{
+		Client:             fakeRemoteClient,
+		ClusterManager:     mockManager, // Ok to use a mock as long the remoteCommonArea.StartWatching is not tested
+		ClusterSetID:       "clusterSetA",
+		ClusterID:          "leaderA",
+		localClusterID:     "clusterA",
+		config:             nil, // Not used for this test
+		scheme:             scheme,
+		Namespace:          "cluster-a-ns",
+		connected:          false,
+		localClusterClient: nil, // Not used for this test
+	}
+
+	remoteCommonAreaUnderTest.Start()
+
+	done := make(chan bool)
+
+	go func() {
+		// Test that member announce is written to the fakeRemoteClient
+		ctx := context.Background()
+		memberAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
+
+		for i := 0; i < 10; i++ {
+			time.Sleep(1 * time.Second)
+
+			err := fakeRemoteClient.List(ctx, memberAnnounceList, client.InNamespace("cluster-a-ns"))
+			if err != nil {
+				klog.InfoS("member announce not written to remote cluster %v yet", err)
+				continue
+			}
+
+			if !remoteCommonAreaUnderTest.IsConnected() {
+				klog.InfoS("Remote cluster not marked as connected yet")
+				continue
+			}
+			done <- true
+		}
+	}()
+
+	select {
+	case <-done:
+	case <-time.After(10 * time.Second):
```

Comment:
I think this should be longer, there is no point to have 10 times loop with 1 second sleep. Or you can reduce the loop number.